### PR TITLE
chore(function): Cleanup and performance upgrade on notifications

### PR DESCRIPTION
**Pull request by BouVid**
- [ ] New feature
- [ ] Bug fix
- [ ] High impact

**Description of work:**
- Code cleanup.
- Some rewrite of the adaptive card builder.
- API calls made async.
- Lists of API calls is now processed in multiple threads.
- Function-timeout period change from 5 to 10 min.

**Testing:**
- [ ] Can be tested
- [ ] Automatic tests created / updated
- [ ] Local tests are passing

**Checklist:**
- [ ] Considered automated tests
- [ ] Considered updating specification / documentation
- [ ] Considered work items 
- [ ] Considered security
- [ ] Performed developer testing
- [ ] Checklist finalized / ready for review

### CODE REVIEW

File: src/backend/function/Fusion.Resources.Functions/Functions/Notifications/ScheduledReportContentBuilderFunction.cs [hunk 1]
```diff
@@ -95,84 +95,79 @@ private async Task BuildContentForTaskOwner(Guid azureUniqueId)
...
+        // Allocations ending soon with no future allocation
         var listOfPersonnelWithoutFutureAllocations = FilterPersonnelWithoutFutureAllocations(personnelForDepartment);
 
-        //11.Personnel with more than 100 % workload
-        var listOfPersonnelWithTBEContent =
-            personnelForDepartment.Select(p => CreatePersonnelContentWithTotalWorkload(p));
-        var listOfPersonnelWithTBEContentOnlyMoreThan100PercentWorkload = listOfPersonnelWithTBEContent.Where(p => p.TotalWorkload > 100);
-
+        // Personnel with more than 100 % workload
+        var personnelEndingWithNoFutureAllocations = personnelForDepartment
```
Comment (aleklundeq) [inline src/backend/function/Fusion.Resources.Functions/Functions/Notifications/ScheduledReportContentBuilderFunction.cs:150]: This looks like the wrong variablename. Should be "listOfPersonnelWithTBEContent" or something like that...
"personnelEndingWithNoFutureAllocations" relates to the previous point (10)

File: src/backend/function/Fusion.Resources.Functions/Functions/Notifications/ScheduledReportContentBuilderFunction.cs [hunk 2]
```diff
@@ -95,84 +95,79 @@ private async Task BuildContentForTaskOwner(Guid azureUniqueId)
...
-        //4.Open request (no proposedPerson)
+        // Open request (no proposedPerson)
         var totalNumberOfOpenRequests = departmentRequests.Count(req =>
             req.State != null && req.Type != null && !req.Type.Equals("ResourceOwnerChange") &&
             !req.HasProposedPerson &&
             !req.State.Equals("completed", StringComparison.OrdinalIgnoreCase));
 
 
-        //5.Requests with start-date < 3 months
+        // Requests with start-date greater than three months form today
```
Comment (aleklundeq) [inline src/backend/function/Fusion.Resources.Functions/Functions/Notifications/ScheduledReportContentBuilderFunction.cs:120]: nitpick: form = from

File: src/backend/function/Fusion.Resources.Functions/Functions/Notifications/ScheduledReportContentBuilderFunction.cs [hunk 3]
```diff
@@ -95,84 +95,79 @@ private async Task BuildContentForTaskOwner(Guid azureUniqueId)
...
             .Count(x => x.OrgPositionInstance != null &&
                         x.State != null &&
                         !x.State.Equals("completed") && !x.Type.Equals("ResourceOwnerChange") &&
-                        (x.OrgPositionInstance.AppliesFrom < threeMonthsFuture &&
+                        (x.OrgPositionInstance.AppliesFrom < threeMonthsFromToday &&
                          x.OrgPositionInstance.AppliesFrom > today) && !x.HasProposedPerson);
 
 
-        //6.Requests with start-date > 3 months:
+        // Requests with start-date less than three months form today
```
Comment (aleklundeq) [inline src/backend/function/Fusion.Resources.Functions/Functions/Notifications/ScheduledReportContentBuilderFunction.cs:129]: nitpick: form = from

Comment (BouVid) [reply src/backend/function/Fusion.Resources.Functions/Functions/Notifications/ScheduledReportContentBuilderFunction.cs:129]: Good catch!

File: src/backend/function/Fusion.Resources.Functions/Functions/Notifications/ScheduledReportContentBuilderFunction.cs (hunk 2 above)
Comment (BouVid) [reply src/backend/function/Fusion.Resources.Functions/Functions/Notifications/ScheduledReportContentBuilderFunction.cs:120]: Good catch!

**Review by aleklundeq — APPROVED** [review]

**Pull request merged**